tools/src/bin/collect.rs

```diff
@@ -362,9 +364,19 @@ impl Options {
             (None, None, Some(hpke_config_json_path)) => {
                 let reader =
                     File::open(hpke_config_json_path).context("could not open HPKE config file")?;
-                let divviup_hpke_config: DivviUpHpkeConfig =
-                    serde_json::from_reader(reader).context("could not parse HPKE config file")?;
-                HpkeKeypair::try_from(divviup_hpke_config).context("could not convert HPKE config")
+
+                // Divvi Up credential files come with multiple different top level JSON objects.
+                // Iterate over each one and return the first one that is valid.
+                match Deserializer::from_reader(reader)
+                    .into_iter::<Value>()
+                    .collect::<Result<Vec<_>, _>>()?
+                    .into_iter()
```

Comment:
Are these two lines necessary? IIUC you get an `Iterator<Item = Value>` from the first call to `into_iter`, then collect it into `Vec<Value>`, then get another `Iterator<Item = Value>` from the second `into_iter`.

Reply:
Hmm, I constructed this nasty iterator with Friday brain, let me take another stab at it.

Reply:
From `StreamDeserializer`, you're actually getting an `Iterator<Item = Result<Value>>`, where the Result is the result of deserializing the next Value. It's tricky to code-golf this into a neat iterator, but I landed on:
```
                match Deserializer::from_reader(reader)
                    .into_iter::<Value>()
                    .find_map(|partial| DivviUpHpkeConfig::deserialize(partial.ok()?).ok())
```
This swallows an error if the JSON is malformed, but serde gives bad error messages anyway. The supplied file should also be tool-generated, so if there's something wrong with it there isn't any action for the user to take other than opening a bug report.